### PR TITLE
Transfer Tonnage Available Tonnage

### PR DESCRIPTION
**Pull request by NateHogan**

### CODE REVIEW

File: src/EA.Weee.Core/AatfEvidence/EvidenceTonnageData.cs [hunk 1]
```diff
@@ -6,12 +6,44 @@
     [Serializable]
     public class EvidenceTonnageData : EvidenceTonnageDataBase
     {
-        public EvidenceTonnageData(Guid id, WeeeCategory categoryId, decimal? received, decimal? reused)
+        public decimal? AvailableReceived
+        {
+            get 
+            {
+                var availble = (Received - TransferredReceived);
+                return availble < 0 ? 0 : availble;
```
Comment (JohnPadgettCivica) [inline src/EA.Weee.Core/AatfEvidence/EvidenceTonnageData.cs:14]: minor spelling mistake

File: src/EA.Weee.RequestHandlers/Mappings/EvidenceNoteMap.cs [hunk 2]
```diff
@@ -38,7 +38,9 @@ public EvidenceNoteData Map(Note source)
                 Protocol = source.Protocol.HasValue ? (Protocol?)source.Protocol.Value : null,
                 WasteType = source.WasteType.HasValue ? (WasteType?)source.WasteType.Value : null,
                 EvidenceTonnageData = source.NoteTonnage.Select(t =>
-                    new EvidenceTonnageData(t.Id, (WeeeCategory)t.CategoryId, t.Received, t.Reused)).ToList(),
+                    new EvidenceTonnageData(t.Id, (WeeeCategory)t.CategoryId, t.Received, t.Reused,
+                    t.NoteTransferTonnage != null ? t.NoteTransferTonnage.Where(ntt => ntt.TransferNote.Status.Value != Domain.Evidence.NoteStatus.Rejected.Value && ntt.TransferNote.Status.Value != Domain.Evidence.NoteStatus.Void.Value).Select(ntt => ntt.Received).Sum() : null,
```
Comment (JohnPadgettCivica) [inline src/EA.Weee.RequestHandlers/Mappings/EvidenceNoteMap.cs:42]: tempted to put the Rejected and voided status into a local list and then do something like !excludeStatus.Contains(transferNote.Status.Value)

**Comment by JohnPadgettCivica** [comment]
Assume await transferTonnagesValidator.Validate(request.TransferValues) will be re-instated in CreateTransferEvidenceNoteRequestHandler? and that the transferTonnagesValidator will be updated to include the correct status...

File: src/EA.Weee.RequestHandlers/Mappings/EvidenceNoteMap.cs (hunk 2 above)
Comment (meglenapnovoselska) [inline src/EA.Weee.RequestHandlers/Mappings/EvidenceNoteMap.cs:42]: have you added tests for showing how the sum gets calculate?

Comment (meglenapnovoselska) [reply src/EA.Weee.RequestHandlers/Mappings/EvidenceNoteMap.cs:42]: or that will be done later?

**Review by meglenapnovoselska — APPROVED** [review]

File: src/EA.Weee.RequestHandlers/Mappings/EvidenceNoteMap.cs (hunk 2 above)
Comment (NateHogan) [reply src/EA.Weee.RequestHandlers/Mappings/EvidenceNoteMap.cs:42]: Tests will be added later

File: src/EA.Weee.RequestHandlers/AatfEvidence/TransferTonnagesValidator.cs [hunk 3]
```diff
@@ -34,7 +34,7 @@ await evidenceDataAccess.GetTonnageByIds(transferValues.Select(t => t.TransferTo
                 //TODO: this needs to be updated to only exclude rejected, voided notes
```
Comment (JohnPadgettCivica) [inline src/EA.Weee.RequestHandlers/AatfEvidence/TransferTonnagesValidator.cs:34]: Remove this comment... I also think there are unit test / integration tests that need to be re-instated. Search for "Re-instate when available tonnages i" in the code

Comment (NateHogan) [reply src/EA.Weee.RequestHandlers/AatfEvidence/TransferTonnagesValidator.cs]: I believe this is done now

**Pull request merged**